sympy/combinatorics/fp_groups.py

```diff
@@ -514,6 +514,138 @@ def elements(self):
+    homomorphism -- Return a homomorphism whenever `homomorphism` = True
+
+    Returns
+    =======
+    * When `homomorphism` = True, quotient group along with the homomorphism
+    from the quotient to the parent_group whose image is isomorphic to G
+    when G is generated any set of elements of the parent_group
+    * Only the quotient group in all other cases.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.fp_groups import FpGroup
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> from sympy.combinatorics.fp_groups import subgroup_quotient, maximal_abelian_quotient
+
+    >>> F, x, y = free_group("x, y")
+    >>> f = FpGroup(F, [x**2, y**3, (x*y)**4])
+    >>> T = subgroup_quotient(f, [x, y])
+    >>> H = f.subgroup([x, y])
+    >>> T.order() == f.order()/H.order()
+    True
+
+    >>> f = FpGroup(F, [x**2, y**3, (x*y)**4])
+    >>> T = subgroup_quotient(f, [x*y**2*x*y, y**2*x*y*x, y**-1])
+    >>> H = f.subgroup([x*y**2*x*y, y**2*x*y*x, y**-1])
+    >>> T.order() == f.order()/H.order()
+    True
+
+    >>> G = [x, y]
+    >>> H = [x*y**2*x*y, y**2*x*y*x, x*y*x]
+    >>> T = subgroup_quotient(G, H, parent_group=f)
+    >>> G = f.subgroup(G)
+    >>> H = f.subgroup(H)
+    >>> T.order() == G.order()/H.order()
+    True
+
+    '''
+    def _get_pres(parent_group, F):
+        from sympy.combinatorics.homomorphisms import homomorphism
+        if isinstance(F, list):
+            K, T = parent_group.subgroup(F, homomorphism=True)
+            f_gens = T(K.generators)
+            f_rels = T(K.relators)
+        else:
+            f_gens = F.generators
+            f_rels = F.relators
+        return f_gens, f_rels
+
+    def _check(F):
+        if ((isinstance(F, list) and not all(elem in free_group for elem in F))
+        or (isinstance(F, FpGroup) and not (F.free_group == free_group))):
+            raise ValueError("The group elements must belong to the parent group")
+
+    # If no parent group is specified,
+    # G is set to the parent `FpGroup`
+    if not parent_group:
+        if isinstance(G, list):
+            raise ValueError("The parent_group must be"
+                            "defined when the group is a list")
+        parent_group = G
+
+    free_group = parent_group.free_group
+
+    if not isinstance(parent_group, FpGroup):
+        raise ValueError("The parent group must be an instance"
+                                    "of FpGroup")
+
+    _check(G)
+    _check(H)
+
+    h_gens, h_rels = _get_pres(parent_group, H)
+    T = None
+    # If `G` is generated by any set of elements of the
+    # `parent_group` return the `FpGroup` on a new presentation.
+    if isinstance(G, list):
+        G, T = parent_group.subgroup(G, homomorphism=True)
+        h_gens = T.invert(h_gens)
+        h_rels = T.invert(h_rels)
+        free_group = G.free_group
+
+    g_gens, g_rels = _get_pres(parent_group, G)
```

Comment:
Whatever `G` is at the start, by this point it's already an `FpGroup` because of the previous `if` statement so there is no need for `_get_pres` to take `parent_group` as an argument